provider/pkg/provider/invoke_helm_template.go

```diff
@@ -0,0 +1,193 @@
+	Prov        bool   `json:"prov,omitempty"`
+	Repo        string `json:"repo,omitempty"`
+	UntarDir    string `json:"untar_dir,omitempty"`
+	Username    string `json:"username,omitempty"`
+	Verify      bool   `json:"verify,omitempty"`
+	Version     string `json:"version,omitempty"`
+}
+
+type HelmChartOpts struct {
+	HelmFetchOpts `json:"fetch_opts,omitempty"`
+
+	APIVersions []string               `json:"api_versions,omitempty"`
+	Chart       string                 `json:"chart,omitempty"`
+	Namespace   string                 `json:"namespace,omitempty"`
+	Path        string                 `json:"path,omitempty"`
+	ReleaseName string                 `json:"release_name,omitempty"`
+	Repo        string                 `json:"repo,omitempty"`
+	Values      map[string]interface{} `json:"values,omitempty"`
+	Version     string                 `json:"version,omitempty"`
+}
+
+// helmTemplate performs Helm fetch/pull + template operations and returns the resulting YAML manifest based on the
+// provided chart options.
+func helmTemplate(opts HelmChartOpts) (string, error) {
+	tempDir, err := ioutil.TempDir("", "helm")
+	if err != nil {
+		return "", err
+	}
+	defer os.RemoveAll(tempDir)
+
+	chart := &chart{
+		opts:     opts,
+		chartDir: tempDir,
+	}
+
+	// If the 'home' option is specified, set the HELM_HOME env var for the duration of the invoke and then reset it
+	// to its previous state.
+	if chart.opts.Home != "" {
+		if helmHome, ok := os.LookupEnv("HELM_HOME"); ok {
+			chart.helmHome = &helmHome
+		}
+		err := os.Setenv("HELM_HOME", chart.opts.Home)
+		if err != nil {
+			return "", pkgerrors.Wrap(err, "failed to set HELM_HOME")
+		}
+		defer func() {
+			if chart.helmHome != nil {
+				_ = os.Setenv("HELM_HOME", *chart.helmHome)
+			} else {
+				_ = os.Unsetenv("HELM_HOME")
+			}
+		}()
+	}
+
+	// If Path is set, use a local Chart, otherwise fetch from a remote.
+	if len(chart.opts.Path) > 0 {
+		chart.chartDir = chart.opts.Path
+	} else {
+		err = chart.fetch()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	result, err := chart.template()
+	if err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+type chart struct {
+	opts      HelmChartOpts
+	chartDir  string
+	chartName string
+	helmHome  *string // Previous setting of HELM_HOME env var (if any)
+}
+
+// fetch runs the `helm fetch` action to fetch a Chart from a remote URL.
+func (c *chart) fetch() error {
+	p := action.NewPull()
+	p.Settings = cli.New()
+	p.CaFile = c.opts.CAFile
+	p.CertFile = c.opts.CertFile
+	p.DestDir = c.chartDir
+	//p.DestDir = c.opts.Destination // TODO: Not currently used, but might be useful for caching
+	p.KeyFile = c.opts.KeyFile
+	p.Keyring = c.opts.Keyring
+	p.Password = c.opts.Password
+	// c.opts.Prov is unused
+	p.RepoURL = c.opts.HelmFetchOpts.Repo
+	p.Untar = true
+	p.UntarDir = c.chartDir
+	p.Username = c.opts.Username
+	p.Verify = c.opts.Verify
+
+	if c.opts.Version == "" && c.opts.Devel {
```

Comment:
on line 97 you checked the length of the string, here we're checking for a blank string? any reason for the difference?

Reply:
I think I copied this line from the helm CLI. I'll update it to match